pkg/liquidity-mining/test/DistributionScheduler.test.ts

```diff
@@ -0,0 +1,394 @@
+
+              it('writes the expected node', async () => {
+                const oldRewardNode = await getRewardNode(insertedTime);
+                expect(oldRewardNode.amount).to.be.eq(0);
+
+                await scheduleDistribution(1, insertedTime);
+
+                const newRewardNode = await getRewardNode(insertedTime);
+                expect(newRewardNode.amount).to.be.eq(1);
+                expect(newRewardNode.nextNodeKey).to.be.eq(NULL);
+              });
+            });
+
+            context('when a reward is being inserted in between two existing rewards', () => {
+              let insertedTime: BigNumber;
+
+              sharedBeforeEach('set insertedTime', async () => {
+                insertedTime = startTime.add(3 * WEEK);
+              });
+
+              it('updates the previous node to point at the new node', async () => {
+                const prevNodeKey = insertedTime.sub(WEEK);
+                expect(await getNextNodeKey(prevNodeKey)).to.be.eq(insertedTime.add(WEEK));
+
+                await scheduleDistribution(1, insertedTime);
+
+                expect(await getNextNodeKey(prevNodeKey)).to.be.eq(insertedTime);
+              });
+
+              it('writes the expected node', async () => {
+                const oldRewardNode = await getRewardNode(insertedTime);
+                expect(oldRewardNode.amount).to.be.eq(0);
+
+                await scheduleDistribution(1, insertedTime);
+
+                const newRewardNode = await getRewardNode(insertedTime);
+                expect(newRewardNode.amount).to.be.eq(1);
+                expect(newRewardNode.nextNodeKey).to.be.eq(insertedTime.add(WEEK));
+              });
+            });
+
+            context('when a reward is being added onto an existing reward', () => {
+              let insertedTime: BigNumber;
+
+              sharedBeforeEach('set insertedTime', async () => {
+                insertedTime = startTime.add(2 * WEEK);
+              });
+
+              it('maintains the existing link from the previous node', async () => {
+                const prevNodeKey = insertedTime.sub(2 * WEEK);
+                expect(await getNextNodeKey(prevNodeKey)).to.be.eq(insertedTime);
+
+                await scheduleDistribution(1, insertedTime);
+
+                expect(await getNextNodeKey(prevNodeKey)).to.be.eq(insertedTime);
+              });
+
+              it('writes the expected node', async () => {
+                const oldRewardNode = await getRewardNode(insertedTime);
+                expect(oldRewardNode.amount).to.be.eq(100);
+
+                await scheduleDistribution(1, insertedTime);
+
+                const newRewardNode = await getRewardNode(insertedTime);
+                expect(newRewardNode.amount).to.be.eq(101);
+                expect(newRewardNode.nextNodeKey).to.be.eq(oldRewardNode.nextNodeKey);
+              });
+
+              context('when providing more tokens would cause an overflow on the node', () => {
```

Comment:
I wrote a note on my notebook to check if this test case existed while reviewing the code. Well done :grin: